warehouse/oidc/views.py

```diff
@@ -68,81 +96,119 @@ def oidc_audience(request):
     require_csrf=False,
     has_translations=True,
 )
-def mint_token_from_oidc(request):
-    def _invalid(errors):
-        request.response.status = 422
-        return {"message": "Token request failed", "errors": errors}
+@view_config(
+    route_name="oidc.mint_token",
+    require_methods=["POST"],
+    renderer="json",
+    require_csrf=False,
+    has_translations=True,
+)
+def mint_token_from_oidc(request: Request):
+    try:
+        payload = TokenPayload.model_validate_json(request.body)
+        unverified_jwt = payload.token
+    except ValidationError as exc:
+        return _invalid(
+            errors=[{"code": "invalid-payload", "description": str(exc)}],
+            request=request,
+        )
 
-    if request.flags.disallow_oidc(AdminFlagValue.DISALLOW_GITHUB_OIDC):
+    # We currently have an **unverified** JWT. To verify it, we need to
+    # know which OIDC service's keyring to check it against.
+    # To do this, we gingerly peek into the unverified claims and
+    # use the `iss` to key into the right `OIDCPublisherService`.
+    try:
+        unverified_claims = jwt.decode(
+            unverified_jwt, options=dict(verify_signature=False)
+        )
+        unverified_issuer: str = unverified_claims["iss"]
+    except Exception:
```

Comment:
Can we make this a more specific set of exceptions?

Reply:
Yeah, I think we could probably make this `(KeyError, PyJWTError)`, although we'll probably have the same exception leakiness we saw in `verify_jwt_signature`:

https://github.com/pypi/warehouse/blob/de4062b6b27f9386e038ee8777e2afbbfdf2cca6/warehouse/oidc/services.py#L223-L240

Reply:
@woodruffw I think you forgot to address this.

Reply:
(Or maybe your comment was actually a question about whether we should do that instead, IMO yes)

Reply:
Yep, was meant to be a comment, sorry for the ambiguity! I'll do this now.

Reply:
Done in [c6a8d96](https://github.com/pypi/warehouse/pull/15148/commits/c6a8d965ef9b367dfabf53d898cf0f3dbac6b604) -- we still need to catch the generic `Exception` because `jwt.decode` might be leaky, but we specialize its handling in the same way mentioned above.